mcr.go

```diff
@@ -154,6 +154,11 @@ func (svc *MCRServiceOp) BuyMCR(ctx context.Context, req *BuyMCRRequest) (*BuyMC
 		order,
 	}
 
+	validateErr := svc.Client.ProductService.ValidateProductOrder(ctx, mcrOrders)
+	if validateErr != nil {
+		return nil, validateErr
+	}
+
```

Comment:
I don't think we should validate in the context of the `Buy[Product]()` request because it makes it harder for a user to understand if their request failed validation or another issue. I would add a method to each service like `ValidateMCR` that wraps the product service call.